test/cuda_sims.jl

```diff
@@ -0,0 +1,344 @@
+  cpu_soln = solve(prob, Tsit5(), save_everystep=false, save_idxs=[:n, :w]);
+
+  # CUDA Setup and Solve
+  cuda_sim = eval(gensim(Klausmeier, dimension=1, code_target=cuda()))
+  cuda_lap_mat = CuSparseMatrixCSC(lap_mat)
+  function cuda_generate(sd, my_symbol; hodge=DiagonalHodge())
+    op = @match my_symbol begin
+      :Δ => x -> begin
+      cuda_lap_mat * x
+      end
+    end
+    return op
+  end
+
+  cuda_fₘ = cuda_sim(sd, cuda_generate, DiagonalHodge())
+
+  cuda_u₀ = ComponentArray(n = CuArray(n), w = CuArray(w), dX = CuArray(dX))
+
+  prob = ODEProblem(cuda_fₘ, cuda_u₀, (0.0, tₑ), constants_and_parameters)
+  cuda_soln = solve(prob, Tsit5(), save_everystep=false, save_idxs=[:n, :w]);
+
+  @test all(isapprox(cpu_soln(tₑ).n, Array(cuda_soln(tₑ).n); atol=1e-11))
+  @test RMSE(cpu_soln(tₑ).n, Array(cuda_soln(tₑ).n)) < 1e-13
+end
+
+@testset "Klausmeier Float32" begin
+  Klausmeier = @decapode begin
+    (n,w)::DualForm0
+    dX::Form1
+    (a,ν,m)::Constant
+  
+    ∂ₜ(w) == a - w - w * n^2 + ν * L(dX, w)
+    ∂ₜ(n) == w * n^2 - m*n + Δ(n)
+  end
+  
+  Klausmeier[9, :type] = :DualForm0
+  Klausmeier[10, :type] = :DualForm0
+  Klausmeier[15, :type] = :DualForm0
+
+  constants_and_parameters = (m = 0.45f0,
+         a = 0.94f0,
+         ν = 182.5f0)
+
+  tₑ = 3.0f0
+
+  function circle(n, c)
+    s = EmbeddedDeltaSet1D{Bool, Point2D}()
+    map(range(0, 2pi - (pi/(2^(n-1))); step=pi/(2^(n-1)))) do t
+      add_vertex!(s, point=Point2D(cos(t),sin(t))*(c/2pi))
+    end
+    add_edges!(s, 1:(nv(s)-1), 2:nv(s))
+    add_edge!(s, nv(s), 1)
+    sd = EmbeddedDeltaDualComplex1D{Bool, Float32, Point2{Float32}}(s)
+    subdivide_duals!(sd, Circumcenter())
+    s,sd
+  end
+  s,sd = circle(13, 500)
+
+  n_dist = Distributions.Normal(pi)
+  n = Float32.([pdf(n_dist, t)*(√(2pi))*7.2 + 0.08 - 5e-2 for t in range(0,2pi; length=ne(sd))])
+
+  w_dist = Distributions.Normal(pi, 20)
+  w = Float32.([pdf(w_dist, t) for t in range(0,2pi; length=ne(sd))])
+
+  dX = sd[:length]
+
+  lap_mat = dec_hodge_star(1,sd) * dec_differential(0,sd) * dec_inv_hodge_star(0,sd) * dec_dual_derivative(0,sd)
+
+  # CPU Setup and Solve
+  cpu_sim = eval(gensim(Klausmeier, dimension=1, stateeltype=Float32))
+
+  function cpu_generate(sd, my_symbol; hodge=DiagonalHodge())
+    op = @match my_symbol begin
+      :Δ => x -> begin
+        lap_mat * x
+      end
+    end
+    return op
+  end
+
+  cpu_fₘ = cpu_sim(sd, cpu_generate, DiagonalHodge())
+
+  cpu_u₀ = ComponentArray(n = n, w = w, dX = dX)
+
+  prob = ODEProblem(cpu_fₘ, cpu_u₀, (0.0, tₑ), constants_and_parameters)
+  cpu_soln = solve(prob, Tsit5(), save_everystep=false, save_idxs=[:n, :w]);
+
+  # CUDA Setup and Solve
+  cuda_sim = eval(gensim(Klausmeier, dimension=1, code_target=cuda(), stateeltype=Float32))
+  cuda_lap_mat = CuSparseMatrixCSC(lap_mat)
+  function cuda_generate(sd, my_symbol; hodge=DiagonalHodge())
+    op = @match my_symbol begin
+      :Δ => x -> begin
```

Comment:
Is this included now

Reply:
No, this is a dual-dual Laplace so this is not in ```default``` set. This should however be an issue, along with other operators.